### ISSUE TITLE
18EU Minor Powers Variant

### ISSUE THREAD

**Issue by ryandriskel** [tobymao/18xx#6929]
Split from https://github.com/tobymao/18xx/issues/792

May be worth implementing the Minor Powers Variants as well:

- [ ] Draft of initial powers (can be coded as privates?) instead of auction procedure
- [ ] Draft of start locations for privates from the available spaces in power-number-order.
- [ ] Slightly different Protected Hexes
- [ ] Implement minor powers (some might be hard).
- [ ] Minor powers follow minor into major when forming majors (hence private implementation?)
- [ ] Sixteenth space (hamburg) used as start spot in 4p games only.

**Comment by ryandriskel** [comment]
For a 2-player game, one optional suggestion – not mentioned in these rules as originally posted – is to apply the changes specified below for a 4-player game so each player receives the same number of Minor Companies. With 6 players, even distribution can be achieved by using 12 Minor Companies, leaving those not selected out of the game.

**Setup and game play differences**
Minor Companies do not start in cities as marked on the board in original 18EU. Instead they have special powers, and start later in any open city. The major companies retain the powers of the Minor Companies that merge into them.
Players start the game with no money.

Instead of bidding for Minor Companies, players simply take them. Each player takes one Minor Company, then the priority is passed to the left – so the player choosing first in one round has the last choice in the next round. This progression includes the final round of taking Minor Companies and thus determines the priority for the first regular stock round.
Following the distribution of all 15 Minor Companies (or 16 in a 4-player game; see below), players select a starting location for their Minor Companies in Minor Company number order. Possible starting locations of Minor Companies are the same starting locations as in the original 18EU (with the addition of Hamburg in a 4-player game). The same reserved hexes for Minor Companies are used, plus the hex northeast of Paris for the Minor Company starting in Paris (NE), and Leipzig for the Minor Company starting in Dresden.

A company may not improve the same hex twice in one OR, neither by playing a tile and upgrading it, nor by upgrading it twice. Even when using special upgrade powers, green tiles are not available until the first 3-Train has been purchased. This is also true for brown and grey tiles with their respective train purchases.
All other rules are the same as in the original 18EU.

**Changes for a 4-player game**
For a 4-player game, one additional Minor Company is added: Minor #0, which has no special power. Also, for a 4- player game, Minor #1 gets a special power because it no longer has the advantage of going first.
An additional starting location – Hamburg – is required for a 4-player game. This is the only way to get a token there, and a token in Hamburg does not block the hex for other companies. A Red-to-Red run including Hamburg as an endpoint does get a bonus for this token (if owned by the operating company).

**List of Minor Company Powers**

Minor #0 (used only in a 4-player game)
No special power.

Minor #1 (power only applies in a 4-player game; otherwise no special power)
The company has a Mail Contract that pays £10 in the yellow phase, £20 in the green phase, £30 in the brown phase and £40 in the grey phase. This money is paid directly to the company's treasury during the revenue part of each operating round. This power should only be used in a 4-player game.

Minor #2
Add £20 to port revenue if using a port on any route. This gives a total of £30 for the port.
   
Minor #3
Gets two £60 credits for building on mountain hexes. Both can be used together to build on a £120 Alp (one of the “M” hexes). After both credits are used, all further mountain builds are at half price. If one credit has been used for a £60 mountain, the other may be used to build on a £120 Alp for £30.

Minor #4
Get one additional yellow tile placement per set of operating rounds. This power cannot be used during the first OR of the game.

Minor #5
Allows free passage for one train route through a city that is blocked by other companies’ tokens. Use once per operating round. That city does not count against the train’s limit and does not earn revenue.

Minor #6
Get one additional track (not town or city) upgrade per set of operating rounds. May not improve the same hex twice in one OR.

Minor #7
Once, at any time after this power is in a Corporation, cash equal to the Corporation’s current stock value can be added to the Corporation’s treasury. Discard this card when this power is used.

Minor #8
Get one additional town (not track or city) upgrade per set of operating rounds. May not improve the same hex twice in one OR.

Minor #9
This company receives a 10% discount on the purchase from the bank of any train or Pullman. This discount can only be applied to only one train or Pullman purchase per operating round.

Minor #10
Increase city limit of one train by one city on each operating round. This power may be used for the same or a different train on subsequent operating rounds.

Minor #11
Add £10 in yellow/green phases, or £20 in brown/grey phases, to the value of one city. Count bonus for each route that uses this city. This power may be used for the same or a different city on subsequent operating rounds.

Minor #12
Increase train limit by one train.

Minor #13
Get one additional city (not track or town) upgrade per set of operating rounds. May not improve the same hex twice in one OR.

Minor #14
This company can take an available free Pullman whenever the company is not at its train limit. This power takes effect after the first 3-Train is purchased.

Minor #15
This company can keep its starting 2-Train for the entire game if it has one when the first 4-Train is purchased. After the first 4-Train is purchased, this train cannot be bought or sold, does not count against the company's train limit, may not be run alone and does not prevent a forced train purchase.

**Comment by ryandriskel** [comment]
My implementation suggestion is to leave Minor 1 out in a non-4 player game, as it is simpler than shuffling powers around and achieves the same positioning.
